inorbit_edge/video.py

```diff
@@ -0,0 +1,149 @@
+# This module provides video capturing capabilities. It allows to stream images from
+# cameras, RTSP streams and more (everything support by OpenCV) to the InOrbit Platform.
+#
+# The functionality is split into two kind of classes:
+# * Cameras: Take care of getting frames from a video source, like a webcam, file or
+#   stream.
+# * CameraStreamer: Consumes frames from a camera and send them to the platform.
+#
+# Future improvements / TODOs:
+#   * Honor module states camera settings, like rate, size and quality.
+#   * Decouple CameraStreamer from image processing and move it to robot.py
+#   * Complete type annotations
+
+import logging
+import threading
+import time
+
+try:
+    import cv2
+except Exception:
+    logging.getLogger(__name__).warning(
+        "Could not import cv2. Video support won't work"
+    )
+
+
+class Camera:
+    """Interface that all camera classes must implement"""
+
+    def __init__(self):
+        raise Exception("Abstract class")
```

Comment:
I'd use the ABC module for abstract classes https://docs.python.org/3/library/abc.html

```python
from abc import ABC, abstractmethod
 
class Camera(ABC):
    def __init__(self):
        super().__init__()
    
    @abstractmethod
    def open(self):
        pass

    ...
```

Reply:
I had no idea about it, you rock Lean ! I'll do the change

Reply:
I had no idea about that, super nice !!! fixed